src/typechecker/FStar.TypeChecker.NBE.fs

```diff
@@ -498,6 +499,29 @@ let rec translate (cfg:config) (bs:list<t>) (e:term) : t =
       let cfg = reifying_true cfg in
       translate cfg bs (fst arg)
 
+    | Tm_app({n=Tm_constant (FC.Const_reflect _)}, [arg]) ->
+      mk_t <| Reflect (translate cfg bs (fst arg))
+
+    | Tm_app({n=Tm_fvar fv}, [_])
+         when S.fv_eq_lid fv PC.assert_lid ||
+              S.fv_eq_lid fv PC.assert_norm_lid ->
+      debug (fun () -> BU.print_string "Eliminated assertion\n");
```

Comment:
NBE is CBV. So, if you write `assert p` it will normalize `p`.

This is wasteful, so there's a bit of special handling of assert and assert_norm to just erase the whole thing to ()